### ISSUE TITLE
Work only with certain css files

### ISSUE THREAD

**Issue by Development-IT** [glenn2223/vscode-live-sass-compiler#358]
Hello. How can I configure to work only with certain css files and not with the whole folder?

**Comment by glenn2223** [comment]
Depends in what regard. Can you expand? 

A couple of examples of the to of my head:

If you have partial files that don't confirm to the "norm", you can use the [liveSassCompile.settings.partialsList](https://github.com/glenn2223/vscode-live-sass-compiler/blob/master/docs/settings.md#livesasscompilesettingspartialslist) setting. This says "here are all my partial files, when I save one compile everything else"

Alternatively; if you only have a couple of files - that are probably gigantic, you can use the [liveSassCompile.settings.includeItems](https://github.com/glenn2223/vscode-live-sass-compiler/blob/master/docs/settings.md#livesasscompilesettingsincludeitems) setting. This says "hey, only compile me when I am one of these files"

If this doesn't help, context will help me 😉 - again, can you expand?

**Comment by glenn2223** [comment]
Did you manage to get it to work?

**Comment by Development-IT** [comment]
not really. I only need to compile the files:
/assets2/css/style.min.css
/assets2/css/style.min.css.map
How to implement this feature?

**Comment by glenn2223** [comment]
This extension compiles `.scss` and `.sass` files, not '.css' files. Whats the path and file name for your SASS file(s)?

**Comment by Development-IT** [comment]
The files are located at the address: /assets2/scss then there are various project folders with .scss files. If a new .scss file is added there and an old .scss file is edited, then accordingly they must, in addition to their changes, regenerate /assets2/css/style.min.css and /assets2/css/style.min.css.map This is a requirement of the main programmer of the project. Right now all /assets2/css files are being regenerated I take it this folder is set as default for regeneration?

**Comment by glenn2223** [comment]
It sounds like everything else is just "partial" files (that is, included within others). Provided you have a `/assets2/scss/style.scss` file, the settings you'd need are: (in addition to any others you'd like to have set)

```JS
{
    "liveSassCompile.settings.formats": [
        {
            "format": "expanded",
            "extensionName": ".css",
            "savePath": "/assets2/css"
        },
        {
            "format": "compressed",
            "extensionName": ".min.css",
            "savePath": "/assets2/css"
        }
    ],
    "liveSassCompile.settings.partialsList": [ "**/!(style).scss" ]
}
```

The `formats` section creates an expanded and compressed version, with a ".css" and ".min.css" suffix respectively - both saved to the same folder.

The `partialsList` says that any `.scss` file that isn't called "style.scss" (you can be more specific if required) is a partial file. So when any one of these other files is saved it will compile `styles.scss` into the two formats, detailed above.

----

Does this work?

**Comment by Development-IT** [comment]
No, my project is organized differently. Right now nothing compiles at all. Can you explain step by step how to customize the extension? The instructions given on the application page don't work. Do I need a server or some other specific settings?

**Comment by Development-IT** [comment]
My application settings:
`    "liveSassCompile.settings.autoprefix": [
    ],
    "liveSassCompile.settings.formats": [
        {
            "format": "expanded",
            "extensionName": ".css",
            "savePath": "/assets2/css",
        }
    ],
    "liveSassCompile.settings.compileOnWatch": false,
    "liveServer.settings.ChromeDebuggingAttachment": false,`
    Compilation fails despite changes to the .scss files

**Comment by glenn2223** [comment]
It's impossible to help without seeing your file structure 

If you want to customise the extension, all the settings are at https://github.com/glenn2223/vscode-live-sass-compiler/blob/master/docs/settings.md - there's YouTube videos too 

If compilation is specifically erroring then what's the error message in the output window that pops up. If nothing is popping up the extension might not even be activated. There are many reasons why compilation wouldn't do what you expect.

**Comment by glenn2223** [comment]
I want to help but I'm afraid you're not giving me any useful information. It might also be with checking the why aren't my files compiling faq https://github.com/glenn2223/vscode-live-sass-compiler/blob/master/docs/faqs.md

**Comment by Development-IT** [comment]
Project Structure:
http://joxi.ru/a2XvM59HZlDjpr
As you can see from it, there is a separate folder with scss files, which in turn is subdivided into other folders, with the `.scss` files themselves.
File content `settings.json`
http://joxi.ru/bmojaM1To79G8r
The files should compile `/assets2/css`
But they are not generated even if you activate the corresponding button.
The extension is also on
http://joxi.ru/Y2LvnN9HeMQNwr

**Comment by glenn2223** [comment]
First, you need to have the setting file located at `.vscode/settings.json` - from the root of your workspace.

After that, when you click the button at the bottom to "Start watching", what happens in the output window here:  
![image](https://github.com/glenn2223/vscode-live-sass-compiler/assets/33450392/27be6090-32c9-4110-91b0-15d907c31126)

**Comment by Development-IT** [comment]
You can't just work in the folder, you only need to work through the workspace? Is everything else filled out correctly?

**Comment by Development-IT** [comment]
I managed to achieve compilation, through the creation of a workspace. Thanks for the clarification. One more question. How can I ignore other files in the css folder that get compiled, in particular other .css.map files?

**Comment by glenn2223** [comment]
Just opening a fodler works. It just needs a `.vscode` folder in it with a `settings.json` file in that. If you managed to get it working with a workspace, that's great.



> I managed to achieve compilation, through the creation of a workspace. Thanks for the clarification. One more question. How can I ignore other files in the css folder that get compiled, in particular other .css.map files?

Sorry, I don't understand. Do you not want to create map files at all?

**Comment by Development-IT** [comment]
I have an additional .map file in my /css folder (besides the main one), how can I add it to ignored so that it is ignored when compiling?

**Comment by glenn2223** [comment]
If you don't want the .map file creating then you need to add the below line:

```JS
{
    "liveSassCompile.settings.formats": [
        {
            "format": "expanded",
            "extensionName": ".css",
            "savePath": "/assets2/css",
         // ADD THE LINE BELOW
            "generateMap": false
        }
    ]
}
```

---

Does that help?

**Comment by Development-IT** [comment]
and **exclude** the specific .map file?

**Comment by Development-IT** [comment]
Or remove a specific folder with .scss files from the compilation, which is part of the general scss folder?

**Comment by glenn2223** [comment]
It only compiles from SASS, there's no need for the extension to exclude `.map` files. 

Again, if you want to stop them from being generated the above line will stop generating `.map` files. If you delete the `.map` files it generated historically, they will not come back.

> Or remove a specific folder with .scss files from the compilation, which is part of the general scss folder?

You can find it here: https://github.com/glenn2223/vscode-live-sass-compiler/blob/master/docs/settings.md#livesasscompilesettingsexcludelist

**Comment by Development-IT** [comment]
and what path should I use to exclude a specific folder, from the shared .scss folder?

**Comment by glenn2223** [comment]
You add your path as a third directory, making it become `["/**/node_modules/**", "/.vscode/**", "/your/other/folder/**]`

**Comment by Development-IT** [comment]
The /assets2/scss/work_interface folder still participates in compilation. Is there any way to exclude it, just a separate folder from the general scss folder?

**Comment by Development-IT** [comment]
We managed to find out the reason for unwanted files getting into the compilation. In the _scss_ folder there was a file _work_interface.scss_ which provided generation of unwanted _css_ files. I added _work_interface.scss_ to exceptions and unnecessary files stopped appearing. The final record looked like this:
```
  "liveSassCompile.settings.excludeList": [
            "/assets2/scss/work_interface.scss"
        ]
```
Thank you for your assistance on this issue.

**Comment by glenn2223** [comment]
Glad you got it sorted
